util-core/src/test/scala/com/twitter/io/PipeTest.scala

```diff
@@ -295,23 +300,26 @@ class PipeTest extends FunSuite with Matchers {
     assert(!cf.isDone)
     assertReadEofAndClosed(rw)
   }
+  test("close not satisfied until reads are fulfilled")(closeNotSatisfiedUntillAllReadsDone)
 
-  test("close while read pending") {
+  def closeWhileReadPending = {
```

Comment:
Some of these had to be factored out, for reasons that aren't clear to me, and may be bugs in assert or scalac: it seems to have to do with macros re-evaluating/re-capturing variables that trip up scalac.

Reply:
todo: check if this is still needed in RC3

Reply:
reported at https://github.com/scala/bug/issues/11556

Reply:
I ran into issues with PipeTest also in my previous minor experiment, subscribed to the bug.

Reply:
Factoring the test body into a method consistently worked around for now, if you run into it elsewhere.